### PR TITLE
Add API tests for nvmeof service.ts file

### PR DESCRIPTION
**Pull request by Achintk1491**
Fixes: https://tracker.ceph.com/issues/68551

Signed-off-by: Achint Kaur <ackaur@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts [hunk 1]
```diff
@@ -27,56 +28,156 @@ describe('NvmeofService', () => {
...
+    it('should call getNamespace', () => {
+      service.getNamespace(mockNQN, nsid, mockGroupName).subscribe();
+      const req = httpTesting.expectOne(
+        `api/nvmeof/subsystem/${mockNQN}/namespace/${nsid}?gw_group=${mockGroupName}`
+      );
+      expect(req.request.method).toBe('GET');
+    });
+    it('should call createNamespace', () => {
+      const mockNamespaceObj = {
+        rbd_image_name: 'string',
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts:160]:
```suggestion
        rbd_image_name: 'nvme_ns_image:12345678',
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts [hunk 2]
```diff
@@ -2,13 +2,14 @@ import { TestBed } from '@angular/core/testing';
...
 import { NvmeofService } from '../../shared/api/nvmeof.service';
+import { throwError } from 'rxjs';
 
 describe('NvmeofService', () => {
   let service: NvmeofService;
   let httpTesting: HttpTestingController;
   const mockGroupName = 'default';
   const mockNQN = 'nqn.2001-07.com.ceph:1721041732363';
-
+  const UI_API_PATH = 'ui-api/nvmeof';
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts:12]: lets do the same for API_PATH = 'api/nvmeof'

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts [hunk 3]
```diff
@@ -27,56 +28,156 @@ describe('NvmeofService', () => {
...
   });
 
-  it('should call deleteSubsystem', () => {
-    service.deleteSubsystem(mockNQN, mockGroupName).subscribe();
-    const req = httpTesting.expectOne(`api/nvmeof/subsystem/${mockNQN}?gw_group=${mockGroupName}`);
-    expect(req.request.method).toBe('DELETE');
+  describe('listener', () => {
+    let group = mockGroupName;
+    it('it should listListeners', () => {
+      service.listListeners(mockNQN, group).subscribe();
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts:113]: why this `group` ?
```suggestion
    it('it should listListeners', () => {
      service.listListeners(mockNQN, mockGroupName).subscribe();
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts [hunk 4]
```diff
@@ -27,56 +28,156 @@ describe('NvmeofService', () => {
...
 
-  // initiators
-  it('should call getInitiators', () => {
-    service.getInitiators(mockNQN, mockGroupName).subscribe();
-    const req = httpTesting.expectOne(
-      `api/nvmeof/subsystem/${mockNQN}/host?gw_group=${mockGroupName}`
-    );
-    expect(req.request.method).toBe('GET');
+  describe('namespace', () => {
+    let nsid = '1';
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts:143]:
```suggestion
    const  mockNsid = '1';
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts [hunk 5]
```diff
@@ -27,56 +28,156 @@ describe('NvmeofService', () => {
     expect(service).toBeTruthy();
   });
 
-  // gateways
-  it('should call listGatewayGroups', () => {
-    service.listGatewayGroups().subscribe();
-    const req = httpTesting.expectOne('api/nvmeof/gateway/group');
-    expect(req.request.method).toBe('GET');
-  });
+  describe('gateway', () => {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/nvmeof.service.spec.ts:31]:
```suggestion
  describe(' test gateway APIs', () => {
```
and similarly for others

**Review by afreen23 — COMMENTED** [review]
Thanks @Achintk1491 for the PR.
Some minor comments.

**Comment by afreen23** [comment]
Hey @Achintk1491 I dont see any updates pushed, can you check ?

Thanks

**Comment by Achintk1491** [comment]
> Hey @Achintk1491 I dont see any updates pushed, can you check ?
> 
> Thanks

Hi @afreen23 
Seems some issue with the commit. Sorry about that. I have pushed changes again.

**Review by afreen23 — APPROVED** [review]

**Comment by Achintk1491** [comment]
jenkins test make check

**Pull request merged**